### DISCUSSION TITLE
Create MapValue/dict with string keys and float values

### DISCUSSION THREAD

**Discussion by augcollet** [ibis-project/ibis#10483] [Q&A] [answered]
Hello,

I need your help to resolve a specific problem...

From the following data with postgresql backend : 

```
import ibis
from ibis import _
import os

con = ibis.postgres.connect(
    user=os.getenv('POSTGRES_USER'),
    password=os.getenv('POSTGRES_PASSWORD'),
    host="postgres",
    port=os.getenv('POSTGRES_PORT'),
    database=os.getenv('POSTGRES_DB'),
)
ibis.set_backend(con)

t=ibis.memtable({
    'client_id':[0,1,0,2,3,0,1,2,3],
    'product':['a', 'b', 'a', 'a', 'b', 'c', 'a', 'a', 'b'],
    'amount':[1.2, 2.5, 4.2, 12.7, 1.2, 3.8, 1.4, 3.8, 3],
})
```

![image](https://github.com/user-attachments/assets/cc33dafa-32b7-448f-a3b3-35a6f5263e4b)

I'm trying to perform the following calculation : 
![image](https://github.com/user-attachments/assets/998ead7b-a35d-409e-b1dc-804e7121aa8f)

I tried the following approach : 
```
data=(
    t.group_by(['client_id', 'product'])
    .agg(
        sum_amount=_['amount'].sum()
    )
    .group_by(['client_id'])
    .agg(
        products_and_sum_amounts=ibis.map(
            _['product'].collect(),
            _['sum_amount'].collect()
        )
    )
)
data.execute()
```
I get the following error : 
![image](https://github.com/user-attachments/assets/03b3b9d2-2ccf-4563-888b-e542bdfb97b6)

It seems that ibis uses hstore to store data from a .map, which is incompatible with numeric values.

I have to cast the values ​​to a string before using .collect to get a result.
![image](https://github.com/user-attachments/assets/216e4848-a970-4a1e-b9fe-cd04ff5828b0)

How can I get around this? For example, how can I build a JSON object instead of MapValue?

( My goal is to exploit the resulting pandas dataset to use it with a DictVectorizer under sklearn.
https://scikit-learn.org/1.5/modules/generated/sklearn.feature_extraction.DictVectorizer.html )

Thank you in advance for your support!

**Comment by cpcloud** [comment]
This seems like a thing we might want to change, and we should probably move away from using `hstore` since it's a bit niche. At the time I decided to use it because it was the closest thing to a `map` in postgres (and implementation-wise I think it still is) but JSON is certainly much friendlier to users.

**Reply by cpcloud** [reply]
I can certainly take a look at using json/jsonb internally in the postgres backend to represent maps.

**Comment by cpcloud** [comment, marked as answer]
Closing in favor of #10484.
